### DISCUSSION TITLE
Consider using WASM?

### DISCUSSION THREAD

**Discussion by alexpdp7** [pkgxdev/pkgx#231] [Ideas]
This is likely out of the scope of the project (but maybe it could be added to the FAQ *why* this project will not do this), but I think it would be interesting to provide a WASM runtime and software targetting that runtime.

This would of course reduce performance, but would probably result in very interesting portability (across OSes and even CPUs).

**Comment by jhheider** [comment]
Definitely an option we're considering for future expansion. Totally open if some someone wants to spearhead it.

**Comment by mxcl** [comment]
I'd love to do this, and longer term we will have the resources to explore it. The problem is certainly at least somewhat related to the base parts of the stack not having flexible build systems that can support it so it would be a large amount of man power and time.

WASM at higher levels should be easy enough though, eg. rustc output.

**Comment by plunderbaron** [comment]
Wow good project

**Comment by Paullumere** [comment]
anyone understand how to use?
